kv/kvserver: TestReadLoadMetricAccounting failed

kv/kvserver.TestReadLoadMetricAccounting [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20875529?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20875529?buildTab=artifacts#/) on master @ [72825229532522f6377e706e009c1090eb1f9b23](https://github.com/cockroachdb/cockroach/commits/72825229532522f6377e706e009c1090eb1f9b23):


```
=== RUN   TestReadLoadMetricAccounting
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/645d36459c8d7b7939924714a555a08a/logTestReadLoadMetricAccounting564952480
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:668: DRPC is enabled (via override by TestingGlobalDRPCOption)
    replica_rankings_test.go:441: evaluated [logtags=: *kvpb.AddSSTableRequest
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.111903730,0 TimestampFromServerClock:<nil> Now:1765988944.111903730,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.112240907,0 TimestampFromServerClock:<nil> Now:1765988944.112240907,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.112430629,0 TimestampFromServerClock:<nil> Now:1765988944.112430629,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.112690747,0 TimestampFromServerClock:<nil> Now:1765988944.112690747,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.112914099,0 TimestampFromServerClock:<nil> Now:1765988944.112914099,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.113094298,0 TimestampFromServerClock:<nil> Now:1765988944.113094298,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.113288467,0 TimestampFromServerClock:<nil> Now:1765988944.113288467,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.113495601,0 TimestampFromServerClock:<nil> Now:1765988944.113495601,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.113687234,0 TimestampFromServerClock:<nil> Now:1765988944.113687234,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.114027155,0 TimestampFromServerClock:<nil> Now:1765988944.114027155,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.114296156,0 TimestampFromServerClock:<nil> Now:1765988944.114296156,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.114502831,0 TimestampFromServerClock:<nil> Now:1765988944.114502831,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.114726024,0 TimestampFromServerClock:<nil> Now:1765988944.114726024,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.115037081,0 TimestampFromServerClock:<nil> Now:1765988944.115037081,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.115270173,0 TimestampFromServerClock:<nil> Now:1765988944.115270173,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.115472092,0 TimestampFromServerClock:<nil> Now:1765988944.115472092,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.116546926,0 TimestampFromServerClock:<nil> Now:1765988944.116546926,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.117067138,0 TimestampFromServerClock:<nil> Now:1765988944.117067138,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.117435923,0 TimestampFromServerClock:<nil> Now:1765988944.117435923,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.117959005,0 TimestampFromServerClock:<nil> Now:1765988944.117959005,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.119327656,0 TimestampFromServerClock:<nil> Now:1765988944.119327656,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.122235484,0 TimestampFromServerClock:<nil> Now:1765988944.122235484,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.128790152,0 TimestampFromServerClock:<nil> Now:1765988944.128790152,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.134400195,0 TimestampFromServerClock:<nil> Now:1765988944.134400195,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.143588503,0 TimestampFromServerClock:<nil> Now:1765988944.143588503,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.161041719,0 TimestampFromServerClock:<nil> Now:1765988944.161041719,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.195096870,0 TimestampFromServerClock:<nil> Now:1765988944.195096870,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.RequestLeaseRequest on /Table/Max: header:<key:"\372" > lease:<start:<> replica:<node_id:1 store_id:1 replica_id:1 type:VOTER_FULL > proposed_ts:<wall_time:1765988944261918514 > sequence:1 acquisition_type:Request min_expiration:<wall_time:1765988950077331053 > term:6 > prev_lease:<start:<> expiration:<wall_time:1765988950077331053 > replica:<node_id:1 store_id:1 replica_id:1 type:VOTER_FULL > deprecated_start_stasis:<wall_time:1765988950077331053 > proposed_ts:<wall_time:1765988944077331053 > sequence:1 acquisition_type:Request min_expiration:<> >  {Timestamp:1765988944.262002629,0 TimestampFromServerClock:<nil> Now:0,0 Replica:(n0,s0):? RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:CONSISTENT RoutingPolicy:LEASEHOLDER WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:0 ClientRangeInfo:{DescriptorGeneration:0 LeaseSequence:0 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.LeaseInfoRequest on /Table/Max: header:<key:"\372" >  {Timestamp:1765988944.262738726,0 TimestampFromServerClock:<nil> Now:1765988944.262738726,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:INCONSISTENT RoutingPolicy:NEAREST WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    testcluster.go:1243: lease is now of type: LeaseLeader
    replica_rankings_test.go:536: test #1
    replica_rankings_test.go:444: evaluated [logtags=]: *kvpb.GetRequest on �000000000: header:<key:"\372000000000" > expect_exclusion_since:<>  {Timestamp:1765988944.263133468,0 TimestampFromServerClock:1765988944.263133468,0 Now:1765988944.263106429,0 Replica:(n1,s1):1 RangeID:81 UserPriority:normal Txn:<nil> ReadConsistency:CONSISTENT RoutingPolicy:LEASEHOLDER WaitPolicy:Block LockTimeout:0s MaxSpanRequestKeys:0 TargetBytes:0 WholeRowsOfSize:0 AllowEmpty:false DistinctSpans:false AsyncConsensus:false CanForwardReadTimestamp:false GatewayNodeID:1 ClientRangeInfo:{DescriptorGeneration:1 LeaseSequence:1 ClosedTimestampPolicy:LAG_BY_CLUSTER_SETTING ExplicitlyRequested:false} BoundedStaleness:<nil> TraceInfo:<nil> IndexFetchSpec:<nil> ReturnElasticCPUResumeSpans:false ProfileLabels:[] AmbiguousReplayProtection:false ConnectionClass:default ProxyRangeInfo:<nil> WriteOptions:<nil> DeadlockTimeout:0s HasBufferedAllPrecedingWrites:false IsReverse:false}
    replica_rankings_test.go:209: 
        	Error Trace:	pkg/kv/kvserver/replica_rankings_test.go:209
        	            				pkg/kv/kvserver/replica_rankings_test.go:564
        	Error:      	Max difference between 0 and 85 allowed is 4, but difference was -85
        	Test:       	TestReadLoadMetricAccounting
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/645d36459c8d7b7939924714a555a08a/logTestReadLoadMetricAccounting564952480
--- FAIL: TestReadLoadMetricAccounting (0.71s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReadLoadMetricAccounting.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58004

--- a/pkg/kv/kvserver/replica_rankings_test.go
+++ b/pkg/kv/kvserver/replica_rankings_test.go
@@ -23,11 +23,13 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/load"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/storageutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/logtags"
 	"github.com/stretchr/testify/require"
 )
@@ -468,6 +470,8 @@ func TestReadLoadMetricAccounting(t *testing.T) {
 	// number of keys being read increases, the read bytes scales by keys * 38.
 	const entrySize = 38
 
+	// NB: All test cases must be idempotent (re-runnable) since we use
+	// SucceedsSoon to retry on interference from background activity.
 	testCases := []struct {
 		ba           *kvpb.BatchRequest
 		expectedRQPS float64
@@ -504,35 +508,42 @@ func TestReadLoadMetricAccounting(t *testing.T) {
 
 	for i, testCase := range testCases {
 		t.Logf("test #%d", i+1)
-		// Reset the request counts to 0 before sending to clear previous requests.
-		repl.loadStats.Reset()
+		// Wrap each test case in SucceedsSoon to handle interference from
+		// background activity (e.g., async stats recording from lease upgrades).
+		// If stats don't match expectations, we reset and retry.
+		testutils.SucceedsSoon(t, func() error {
+			// Reset the request counts to 0 before sending to clear previous requests.
+			repl.loadStats.Reset()
 
-		requestsBefore := repl.loadStats.TestingGetSum(load.Requests)
-		writesBefore := repl.loadStats.TestingGetSum(load.WriteKeys)
-		readsBefore := repl.loadStats.TestingGetSum(load.ReadKeys)
-		readBytesBefore := repl.loadStats.TestingGetSum(load.ReadBytes)
-		writeBytesBefore := repl.loadStats.TestingGetSum(load.WriteBytes)
+			_, pErr = db.NonTransactionalSender().Send(ctx, testCase.ba)
+			if pErr != nil {
+				return pErr.GoError()
+			}
 
-		_, pErr = db.NonTransactionalSender().Send(ctx, testCase.ba)
-		require.Nil(t, pErr)
+			requestsAfter := repl.loadStats.TestingGetSum(load.Requests)
+			writesAfter := repl.loadStats.TestingGetSum(load.WriteKeys)
+			readsAfter := repl.loadStats.TestingGetSum(load.ReadKeys)
+			readBytesAfter := repl.loadStats.TestingGetSum(load.ReadBytes)
+			writeBytesAfter := repl.loadStats.TestingGetSum(load.WriteBytes)
 
-		require.Equal(t, 0.0, requestsBefore)
-		require.Equal(t, 0.0, writesBefore)
-		require.Equal(t, 0.0, readsBefore)
-		require.Equal(t, 0.0, writeBytesBefore)
-		require.Equal(t, 0.0, readBytesBefore)
-
-		requestsAfter := repl.loadStats.TestingGetSum(load.Requests)
-		writesAfter := repl.loadStats.TestingGetSum(load.WriteKeys)
-		readsAfter := repl.loadStats.TestingGetSum(load.ReadKeys)
-		readBytesAfter := repl.loadStats.TestingGetSum(load.ReadBytes)
-		writeBytesAfter := repl.loadStats.TestingGetSum(load.WriteBytes)
-
-		assertGreaterThanInDelta(t, testCase.expectedRQPS, requestsAfter, epsilonAllowed)
-		assertGreaterThanInDelta(t, testCase.expectedWPS, writesAfter, epsilonAllowed)
-		assertGreaterThanInDelta(t, testCase.expectedRPS, readsAfter, epsilonAllowed)
-		assertGreaterThanInDelta(t, testCase.expectedWBPS, writeBytesAfter, epsilonAllowed)
-		assertGreaterThanInDelta(t, testCase.expectedRBPS, readBytesAfter, epsilonAllowed)
+			// Check write stats first since that's where interference shows up.
+			if writesAfter < testCase.expectedWPS || writesAfter > testCase.expectedWPS+epsilonAllowed {
+				return errors.Errorf("writeKeys: got %f, expected %f±%d", writesAfter, testCase.expectedWPS, epsilonAllowed)
+			}
+			if writeBytesAfter < testCase.expectedWBPS || writeBytesAfter > testCase.expectedWBPS+epsilonAllowed {
+				return errors.Errorf("writeBytes: got %f, expected %f±%d", writeBytesAfter, testCase.expectedWBPS, epsilonAllowed)
+			}
+			if requestsAfter < testCase.expectedRQPS || requestsAfter > testCase.expectedRQPS+epsilonAllowed {
+				return errors.Errorf("requests: got %f, expected %f±%d", requestsAfter, testCase.expectedRQPS, epsilonAllowed)
+			}
+			if readsAfter < testCase.expectedRPS || readsAfter > testCase.expectedRPS+epsilonAllowed {
+				return errors.Errorf("readKeys: got %f, expected %f±%d", readsAfter, testCase.expectedRPS, epsilonAllowed)
+			}
+			if readBytesAfter < testCase.expectedRBPS || readBytesAfter > testCase.expectedRBPS+epsilonAllowed {
+				return errors.Errorf("readBytes: got %f, expected %f±%d", readBytesAfter, testCase.expectedRBPS, epsilonAllowed)
+			}
+			return nil
+		})
 	}
 }
 

